card/lib/card/spec_helper.rb

```diff
@@ -76,4 +73,45 @@ def render_card_with_args view, card_args={}, format_args={}, view_args={}
   def users
     SharedData::USERS.sort
   end
+
+  # Make expectations in the event phase.
+  # Takes the usual event options :on and :before/:after/:around
+  # and registers the event_block with these options as an event.
+  # Unknown methods in the event_block are executed in the rspec context
+  # instead of the card's context.
+  # An additionaly :trigger block in opts is expected that is called
+  # to start the event phase.
+  # Example:
+  # in_phase before: :approve, on: :save,
+  #          trigger: ->{ test_card.update_attributes! content: '' } do
+  #            expect(item_names).to eq []
+  #          end
+  def in_phase opts, &event_block
+    $rspec_binding = binding
+    Card.class_eval do
+      def method_missing m, *args
+        begin
+          method = eval('method(%s)' % m.inspect, $rspec_binding)
+        rescue NameError
+        else
+          return method.call(*args)
+        end
+        begin
+          value = eval(m.to_s, $rspec_binding)
+          return value
+        rescue NameError
+        end
+        super
```

Comment:
The tricky part to get the right context - either Card or Rspec so that you can use card methods as in usual events but also rspec methods like `expect`.
I couldn't get around the global variable `$rspec_binding` to pass the test's binding into the block. Suggestions for a better solution are welcome.